### PR TITLE
userland: update templates

### PR DESCRIPTION
**Pull request by alarcher**

### CODE REVIEW

File: templates/cmake.mk [hunk 1]
```diff
@@ -26,12 +28,11 @@ COMPONENT_ARCHIVE_URL=
 COMPONENT_ARCHIVE_HASH=
 COMPONENT_LICENSE=
 
-include $(WS_MAKE_RULES)/prep.mk
-include $(WS_MAKE_RULES)/cmake.mk
-include $(WS_MAKE_RULES)/ips.mk
+TEST_TARGET=$(NO_TESTS) # if no testsuite enabled
```
Comment (AndWac) [inline templates/cmake.mk:31]: Does this work as expected? Shouldn't TEST_TARGET only be set if not yet set?

File: templates/ant.mk [hunk 2]
```diff
@@ -26,12 +28,5 @@ COMPONENT_ARCHIVE_URL=
 COMPONENT_ARCHIVE_HASH=
 COMPONENT_LICENSE=
 
-include $(WS_MAKE_RULES)/prep.mk
-include $(WS_MAKE_RULES)/ant.mk
-include $(WS_MAKE_RULES)/ips.mk
+include $(WS_MAKE_RULES)/common.mk
```
Comment (AndWac) [inline templates/ant.mk:31]: What about TEST_TARGET?

File: templates/configure.mk [hunk 3]
```diff
@@ -26,12 +27,10 @@ COMPONENT_ARCHIVE_URL=
 COMPONENT_ARCHIVE_HASH=
 COMPONENT_LICENSE=
 
-include $(WS_MAKE_RULES)/prep.mk
-include $(WS_MAKE_RULES)/configure.mk
-include $(WS_MAKE_RULES)/ips.mk
+TEST_TARGET=$(NO_TESTS) # if no testsuite enabled
```
Comment (AndWac) [inline templates/configure.mk:30]: Again: Set only if not yet done?

File: templates/configure.mk [hunk 4]
```diff
@@ -12,6 +12,7 @@
 # Copyright 2020 <contributor name>
 #
 
+BUILD_BITS=64 # for binaries or 32_and_64 for libraries
```
Comment (AndWac) [inline templates/configure.mk:15]: And what about libraries? Shouldn't we set BUILD_BITS only if not yet done?

File: templates/python.mk [hunk 5]
```diff
@@ -26,12 +28,9 @@ COMPONENT_ARCHIVE_URL=
 COMPONENT_ARCHIVE_HASH=
 COMPONENT_LICENSE=
 
-include $(WS_MAKE_RULES)/prep.mk
-include $(WS_MAKE_RULES)/setup.py.mk
-include $(WS_MAKE_RULES)/ips.mk
+TEST_TARGET=$(NO_TESTS)
```
Comment (AndWac) [inline templates/python.mk]: Unless there aren't any python tests and will never be, shouldn't this also only be set if not yet defined?

File: templates/cmake.mk (hunk 1 above)
Comment (alarcher) [reply templates/cmake.mk:31]: This has the same effect as `test: $(NO_TESTS)` which was previously in the file. Do you intend to change the default?

File: templates/ant.mk (hunk 2 above)
Comment (alarcher) [reply templates/ant.mk:31]: It is already set in common.mk

File: templates/configure.mk (hunk 3 above)
Comment (alarcher) [reply templates/configure.mk:30]: Tests are enabled by default for `configure` recipes.

File: templates/configure.mk (hunk 4 above)
Comment (alarcher) [reply templates/configure.mk:15]: The problem with defaults is that they may not hold anymore... that's why I thougt about adding the comment. Is it the wording that bothers you? Should I write "for executables-only packages or 32_and_64 for libraries"?

File: templates/python.mk (hunk 5 above)
Comment (alarcher) [reply templates/python.mk]: For python packages most are without tests so we could disable them by default.

Comment (alarcher) [reply templates/python.mk]: But then we should fix all the packages using already BUILD_STYLE if needed:
```
narval> ggrep -l "BUILD_STYLE" */Makefile | cut -f 1 -d '/'
argcomplete
argh
attrs
automat
barman
ccsm
constantly
hamcrest
hyperlink
hypothesis
incremental
kafka-python
netifaces
pillow
psutil
psycopg2
pycups
python-dateutil
python-memcached
python-xlib
pyxdg
redis-py
twisted
zope-interface

```

File: templates/cmake.mk (hunk 1 above)
Comment (AndWac) [reply templates/cmake.mk:31]: No, I misunderstood how it works

File: templates/ant.mk (hunk 2 above)
Comment (AndWac) [reply templates/ant.mk:31]: Ah Ok

File: templates/configure.mk (hunk 4 above)
Comment (AndWac) [reply templates/configure.mk:15]: No, it's ok.

**Review by pyhalov — APPROVED** [review]

File: templates/python.mk (hunk 5 above)
Comment (pyhalov) [reply templates/python.mk]: Enabling test target by default is a good practice ;)

Comment (alarcher) [reply templates/python.mk]: OK I will change it then :)

**Pull request merged**